src/ServiceControl.TransportAdapter/FailedMessageForwarder.cs

```diff
@@ -15,44 +16,56 @@ class FailedMessageForwarder<TEndpoint, TServiceControl>
     {
         public FailedMessageForwarder(string adapterName, string frontendErrorQueue, string backendErrorQueue, int retryMessageImmeidateRetries, string poisonMessageQueueName,
             Action<TransportExtensions<TEndpoint>> frontendTransportCustomization,
-            Action<TransportExtensions<TServiceControl>> backendTransportCustomization)
+            Action<TransportExtensions<TServiceControl>> backendTransportCustomization,
+            MetricsContext metricsContext)
         {
-            backEndConfig = RawEndpointConfiguration.Create($"{adapterName}.Retry", (context, _) => OnRetryMessage(context), poisonMessageQueueName);
-            backEndConfig.CustomErrorHandlingPolicy(new RetryForwardingFailurePolicy(backendErrorQueue, retryMessageImmeidateRetries, () => retryToAddress));
+            var errorsForwarded = metricsContext.Meter("Errors forwarded", Unit.Custom("Messages"));
```

Comment:
You mean that the adapter would expose events such as `ErrorForwarded` or `ErrorForwardingFailed` ? I mean, C# events?